src/coreclr/runtime/arm64/WriteBarriers.S

```diff
@@ -229,12 +229,25 @@ LEAF_END RhpByRefAssignRefArm64, _TEXT
         b       C_FUNC(RhpAssignRefArm64)
 
 LOCAL_LABEL(NotInHeap):
+#ifdef TARGET_APPLE
+        b       C_FUNC(RhpCheckedAssignRefArm64_NotInHeap)
+#else
     ALTERNATE_ENTRY RhpCheckedAssignRefAVLocation
         str     x15, [x14], 8
         ret
-
+#endif
 LEAF_END RhpCheckedAssignRefArm64, _TEXT
 
+#ifdef TARGET_APPLE
+// Apple's linker has issues which break unwind info if
+// an ALTERNATE_ENTRY is present in the middle of a function
+LEAF_ENTRY RhpCheckedAssignRefArm64_NotInHeap, _TEXT
+    ALTERNATE_ENTRY RhpCheckedAssignRefAVLocation
+        str     x15, [x14], 8
+        ret
+LEAF_END RhpCheckedAssignRefArm64_NotInHeap, _TEXT
+#endif
```

Comment:
That seems to be a new linker issue previously unknown to me. If you could share the broken .o file and Xcode version I'd appreciate it. The Apple team maintaining the linker seems to be pretty responsive to feedback.

Reply:
I'll make a branch/PR with this problem once I get this whole PR done. I believe I've gotten it to a working state (other than Linux Arm32 correctness), and I don't want to poke it any more if I can avoid it.

Reply:
I'll try to make a small repro but it would be really helpful to know the Xcode version where you experienced this. I've been reporting all these issues to the Apple and they fixed all of them eventually.